File: suc/swift/077.swift
import SwiftUI

@main
struct FurnitureShopApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

struct ContentView: View {
    var body: some View {
        FurnitureDetailScreen()
            .statusBarHidden(true)
    }
}

struct FurnitureDetailScreen: View {
    @State private var isFavorite: Bool = false
    @State private var selectedColorIndex: Int = 0
    @State private var quantity: Int = 1

    let colors: [Color] = [
        Color(red: 0.94, green: 0.71, blue: 0.58), // #EFB495
        Color(red: 0.55, green: 0.43, blue: 0.39), // #8D6E63
        Color(red: 0.36, green: 0.25, blue: 0.22)  // #5D4037
    ]

    var body: some View {
        ZStack {
            // Main Background
            Color(red: 0.96, green: 0.96, blue: 0.97) // #F5F5F7
                .ignoresSafeArea()

            VStack(spacing: 0) {
                ScrollView(showsIndicators: false) {
                    VStack(alignment: .leading, spacing: 0) {
                        
                        // Top Image Section
                        ZStack {
                            // Background shape
                            Color(red: 0.91, green: 0.91, blue: 0.91) // #E8E8E8
                                .frame(height: 400)
                                .clipShape(CustomCornerShape(radius: 40, corners: [.bottomLeft, .bottomRight]))
                            
                            // Centered Icon Container
                            ZStack {
                                Circle()
                                    .fill(Color.white)
                                    .frame(width: 200, height: 200)
                                    .shadow(color: Color.black.opacity(0.2), radius: 20, x: 0, y: 10)
                                
                                Image(systemName: "house.fill")
                                    .resizable()
                                    .aspectRatio(contentMode: .fit)
                                    .frame(width: 100, height: 100)
                                    .foregroundColor(Color(red: 0.94, green: 0.71, blue: 0.58)) // #EFB495
                            }
                            
                            // Top Buttons
                            VStack {
                                HStack {
                                    // Back Button
                                    Button(action: {}) {
                                        Image(systemName: "arrow.backward")
                                            .font(.system(size: 20, weight: .medium))
                                            .foregroundColor(.black)
                                            .frame(width: 42, height: 42)
                                            .background(Color.white)
                                            .clipShape(Circle())
                                    }
                                    
                                    Spacer()
                                    
                                    // Favorite Button
                                    Button(action: { isFavorite.toggle() }) {
                                        Image(systemName: isFavorite ? "heart.fill" : "heart")
                                            .font(.system(size: 20, weight: .medium))
                                            .foregroundColor(isFavorite ? .red : .black)
                                            .frame(width: 42, height: 42)
                                            .background(Color.white)
                                            .clipShape(Circle())
                                    }
                                }
                                .padding(24)
                                Spacer()
                            }
                            .frame(height: 400)
                        }

                        // Content Section
                        VStack(alignment: .leading, spacing: 0) {
                            
                            // Title and Rating
                            HStack(alignment: .center) {
                                Text("Modern Lounge")
                                    .font(.system(size: 26, weight: .bold))
                                    .foregroundColor(Color(red: 0.20, green: 0.20, blue: 0.20)) // #333333
                                
                                Spacer()
                                
                                HStack(spacing: 4) {
                                    Image(systemName: "star.fill")
                                        .resizable()
                                        .frame(width: 20, height: 20)
                                        .foregroundColor(Color(red: 1.0, green: 0.76, blue: 0.03)) // #FFC107
                                    
                                    Text("4.8")
                                        .font(.system(size: 16, weight: .bold))
                                        .foregroundColor(.black)
                                }
                            }
                            
                            // Subtitle
                            Text("Armchair")
                                .font(.system(size: 20, weight: .light))
                                .foregroundColor(.gray)
                                .padding(.top, -4) // Offset to match visual layout
                            
                            Spacer().frame(height: 24)
                            
                            // Description Title
                            Text("Description")
                                .font(.system(size: 18, weight: .semibold))
                                .foregroundColor(.black)
                            
                            Spacer().frame(height: 8)
                            
                            // Description Body
                            Text("Experience the ultimate comfort with this modern minimalist armchair. Crafted with premium bamboo wood and soft linen fabric, it perfectly blends style and functionality for your living room.")
                                .font(.system(size: 14))
                                .foregroundColor(.gray)
                                .lineSpacing(6) // Approx lineHeight 22sp
                            
                            Spacer().frame(height: 24)
                            
                            // Color and Quantity Row
                            HStack(alignment: .top) {
                                // Color Selection
                                VStack(alignment: .leading, spacing: 8) {
                                    Text("Color")
                                        .font(.system(size: 18, weight: .semibold))
                                        .foregroundColor(.black)
                                    
                                    HStack(spacing: 12) {
                                        ForEach(0..<colors.count, id: \.self) { index in
                                            ZStack {
                                                Circle()
                                                    .fill(colors[index])
                                                    .frame(width: 30, height: 30)
                                                
                                                if selectedColorIndex == index {
                                                    Circle()
                                                        .stroke(Color.black, lineWidth: 2)
                                                        .frame(width: 30, height: 30)
                                                }
                                            }
                                            .onTapGesture {
                                                selectedColorIndex = index
                                            }
                                        }
                                    }
                                }
                                
                                Spacer()
                                
                                // Quantity Selection
                                VStack(alignment: .trailing, spacing: 8) {
                                    Text("Quantity")
                                        .font(.system(size: 18, weight: .semibold))
                                        .foregroundColor(.black)
                                    
                                    HStack(alignment: .center, spacing: 0) {
                                        Text("-")
                                            .font(.system(size: 20))
                                            .foregroundColor(.black)
                                            .frame(width: 40, height: 32)
                                            .contentShape(Rectangle())
                                            .onTapGesture {
                                                if quantity > 1 { quantity -= 1 }
                                            }
                                        
                                        Text("\(quantity)")
                                            .font(.system(size: 18, weight: .bold))
                                            .foregroundColor(.black)
                                            .frame(minWidth: 20)
                                        
                                        Image(systemName: "plus")
                                            .font(.system(size: 16))
                                            .foregroundColor(.black)
                                            .frame(width: 40, height: 32)
                                            .contentShape(Rectangle())
                                            .onTapGesture {
                                                quantity += 1
                                            }
                                    }
                                    .background(Color.white)
                                    .cornerRadius(12)
                                }
                            }
                            
                            // Extra padding at bottom of scrollable content to avoid overlap with bottom bar
                            Spacer().frame(height: 100)
                        }
                        .padding(24)
                    }
                }
            }
            
            // Bottom Surface
            VStack {
                Spacer()
                
                HStack {
                    VStack(alignment: .leading, spacing: 2) {
                        Text("Total Price")
                            .font(.system(size: 14))
                            .foregroundColor(.gray)
                        
                        Text("$\(245 * quantity)")
                            .font(.system(size: 24, weight: .bold))
                            .foregroundColor(.black)
                    }
                    
                    Spacer()
                    
                    Button(action: {}) {
                        HStack(spacing: 8) {
                            Image(systemName: "cart.fill")
                                .font(.system(size: 20))
                            Text("Add to Cart")
                                .font(.system(size: 16, weight: .medium))
                        }
                        .foregroundColor(.white)
                        .frame(width: 160, height: 50)
                        .background(Color(red: 0.20, green: 0.20, blue: 0.20)) // #333333
                        .cornerRadius(16)
                    }
                }
                .padding(24)
                .background(Color.white)
                .cornerRadius(24, corners: [.topLeft, .topRight])
                .shadow(color: Color.black.opacity(0.1), radius: 16, x: 0, y: -4)
            }
            .ignoresSafeArea(edges: .bottom)
        }
    }
}

// MARK: - Helper Components

struct CustomCornerShape: Shape {
    var radius: CGFloat = .infinity
    var corners: UIRectCorner = .allCorners

    func path(in rect: CGRect) -> Path {
        let path = UIBezierPath(
            roundedRect: rect,
            byRoundingCorners: corners,
            cornerRadii: CGSize(width: radius, height: radius)
        )
        return Path(path.cgPath)
    }
}

// Helper extension to apply custom corners easily via .cornerRadius modifier wrapper if needed, 
// but strictly keeping it compatible with standard SwiftUI View modifier approach.
extension View {
    func cornerRadius(_ radius: CGFloat, corners: UIRectCorner) -> some View {
        clipShape(CustomCornerShape(radius: radius, corners: corners))
    }
}

File: suc/kotlin/077.kt
package com.example.myapplication

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.FavoriteBorder
import androidx.compose.material.icons.filled.Home
import androidx.compose.material.icons.filled.ShoppingCart
import androidx.compose.material.icons.filled.Star
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        val windowInsetsController = WindowCompat.getInsetsController(window, window.decorView)
        windowInsetsController.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        windowInsetsController.hide(WindowInsetsCompat.Type.systemBars())

        setContent {
            MaterialTheme(
                colorScheme = androidx.compose.material3.lightColorScheme(
                    primary = Color(0xFFEFB495),
                    secondary = Color(0xFF4A403A)
                )
            ) {
                FurnitureDetailScreen()
            }
        }
    }
}

@Composable
fun FurnitureDetailScreen() {
    var isFavorite by remember { mutableStateOf(false) }
    var selectedColorIndex by remember { mutableIntStateOf(0) }
    var quantity by remember { mutableIntStateOf(1) }

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFFF5F5F7))
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(bottom = 80.dp)
        ) {
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(400.dp)
                    .clip(RoundedCornerShape(bottomStart = 40.dp, bottomEnd = 40.dp))
                    .background(Color(0xFFE8E8E8))
            ) {
                Box(
                    modifier = Modifier
                        .align(Alignment.Center)
                        .size(200.dp)
                        .shadow(20.dp, CircleShape)
                        .background(Color.White, CircleShape),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = Icons.Default.Home,
                        contentDescription = null,
                        tint = Color(0xFFEFB495),
                        modifier = Modifier.size(100.dp)
                    )
                }

                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(24.dp),
                    horizontalArrangement = Arrangement.SpaceBetween
                ) {
                    IconButton(
                        onClick = { },
                        modifier = Modifier
                            .background(Color.White, CircleShape)
                            .size(42.dp)
                    ) {
                        Icon(
                            imageVector = Icons.Default.ArrowBack,
                            contentDescription = null,
                            tint = Color.Black
                        )
                    }
                    IconButton(
                        onClick = { isFavorite = !isFavorite },
                        modifier = Modifier
                            .background(Color.White, CircleShape)
                            .size(42.dp)
                    ) {
                        Icon(
                            imageVector = if (isFavorite) Icons.Default.Favorite else Icons.Default.FavoriteBorder,
                            contentDescription = null,
                            tint = if (isFavorite) Color.Red else Color.Black
                        )
                    }
                }
            }

            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(24.dp)
            ) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(
                        text = "Modern Lounge",
                        fontSize = 26.sp,
                        fontWeight = FontWeight.Bold,
                        color = Color(0xFF333333)
                    )
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            imageVector = Icons.Default.Star,
                            contentDescription = null,
                            tint = Color(0xFFFFC107),
                            modifier = Modifier.size(20.dp)
                        )
                        Text(
                            text = "4.8",
                            fontWeight = FontWeight.Bold,
                            fontSize = 16.sp,
                            modifier = Modifier.padding(start = 4.dp)
                        )
                    }
                }

                Text(
                    text = "Armchair",
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Light,
                    color = Color.Gray,
                    modifier = Modifier.offset(y = (-4).dp)
                )

                Spacer(modifier = Modifier.height(24.dp))

                Text(
                    text = "Description",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.Black
                )
                Spacer(modifier = Modifier.height(8.dp))
                Text(
                    text = "Experience the ultimate comfort with this modern minimalist armchair. Crafted with premium bamboo wood and soft linen fabric, it perfectly blends style and functionality for your living room.",
                    fontSize = 14.sp,
                    color = Color.Gray,
                    lineHeight = 22.sp
                )

                Spacer(modifier = Modifier.height(24.dp))

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Column {
                        Text(
                            text = "Color",
                            fontSize = 18.sp,
                            fontWeight = FontWeight.SemiBold
                        )
                        Spacer(modifier = Modifier.height(8.dp))
                        Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                            val colors = listOf(Color(0xFFEFB495), Color(0xFF8D6E63), Color(0xFF5D4037))
                            colors.forEachIndexed { index, color ->
                                Box(
                                    modifier = Modifier
                                        .size(30.dp)
                                        .clip(CircleShape)
                                        .background(color)
                                        .clickable { selectedColorIndex = index }
                                        .border(
                                            width = if (selectedColorIndex == index) 2.dp else 0.dp,
                                            color = if (selectedColorIndex == index) Color.Black else Color.Transparent,
                                            shape = CircleShape
                                        )
                                )
                            }
                        }
                    }

                    Column(horizontalAlignment = Alignment.End) {
                        Text(
                            text = "Quantity",
                            fontSize = 18.sp,
                            fontWeight = FontWeight.SemiBold
                        )
                        Spacer(modifier = Modifier.height(8.dp))
                        Row(
                            verticalAlignment = Alignment.CenterVertically,
                            modifier = Modifier
                                .background(Color.White, RoundedCornerShape(12.dp))
                                .padding(horizontal = 8.dp, vertical = 4.dp)
                        ) {
                            Text(
                                text = "-",
                                fontSize = 20.sp,
                                modifier = Modifier
                                    .padding(horizontal = 8.dp)
                                    .clickable { if (quantity > 1) quantity-- }
                            )
                            Text(
                                text = "$quantity",
                                fontSize = 18.sp,
                                fontWeight = FontWeight.Bold,
                                modifier = Modifier.padding(horizontal = 8.dp)
                            )
                            Icon(
                                imageVector = Icons.Default.Add,
                                contentDescription = null,
                                modifier = Modifier
                                    .size(20.dp)
                                    .clickable { quantity++ }
                            )
                        }
                    }
                }
            }
        }

        Surface(
            modifier = Modifier
                .align(Alignment.BottomCenter)
                .fillMaxWidth(),
            shadowElevation = 16.dp,
            shape = RoundedCornerShape(topStart = 24.dp, topEnd = 24.dp),
            color = Color.White
        ) {
            Row(
                modifier = Modifier
                    .padding(24.dp)
                    .fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column {
                    Text(
                        text = "Total Price",
                        color = Color.Gray,
                        fontSize = 14.sp
                    )
                    Text(
                        text = "$${245 * quantity}",
                        fontSize = 24.sp,
                        fontWeight = FontWeight.Bold,
                        color = Color.Black
                    )
                }

                Button(
                    onClick = { },
                    colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF333333)),
                    shape = RoundedCornerShape(16.dp),
                    modifier = Modifier
                        .height(50.dp)
                        .width(160.dp)
                ) {
                    Icon(
                        imageVector = Icons.Default.ShoppingCart,
                        contentDescription = null,
                        modifier = Modifier.size(20.dp)
                    )
                    Spacer(modifier = Modifier.width(8.dp))
                    Text("Add to Cart")
                }
            }
        }
    }
}